wrapper/src/main/java/software/amazon/jdbc/plugin/failover/ClusterAwareReaderFailoverHandler.java

```diff
@@ -276,9 +276,9 @@ public List<HostSpec> getReaderHostsByPriority(final List<HostSpec> hosts) {
     hostsByPriority.addAll(downHostList);
 
     final int numOfReaders = activeReaders.size() + downHostList.size();
-    if (writerHost != null && (numOfReaders == 0
-          || this.pluginService.getDialect().getFailoverRestrictions()
-                .contains(FailoverRestriction.ENABLE_WRITER_IN_TASK_B))) {
+    final boolean enableWriterInTaskB =
```

Comment:
Just a note.

By resolving value of this expression you disable code optimization. The following expression
may be optimized by compiler so if the first part of expression is true the remaining part is not calculated.

numOfReaders == 0 || enableWriterInTaskB

If you calculate this expression earlier then make the code less performant in some cases.

Reply:
should be fixed now